Translation haogao gu

Koohoko opened the pull request:
Translation from Haogao Gu (group 14).

contents:
_src_ (`zh_CN.po`)
|First line|	Last line|	
|2606|	2807|

_R_ (`R-zh_CN.po`)
|First line|	Last line|
|1264|	1415|

For the src (`zh_CN.po`), there are some chinese characteres (line 3056 - 3252) which have been previously added by other translator, appear to crash into nonsense characters on my machine, I am not sure if it is due to encoding problem of my editor. So I copy them from their previous commit on the web and paste back to my file. :(
Also I have difficult in triggering the message, so in fact I did not actually go through the testing. Sorry about that.

Thank you!

MichaelChirico commented:
Excellent! Pending review from @biobai. I'll try and have a look into the encoding thing. Seems to be affecting everyone 😓

MichaelChirico commented:
Have merged to the `translation` branch which covers the encoding thing. `update_pkg_po` working as intended -- all that's left is to confirm the Chinese itself! :)

Koohoko commented:
@MichaelChirico Thank you!

biobai commented on po/zh_CN.po:
```diff
@@ -3092,48 +3092,48 @@ msgstr ""
 #, c-format
 msgid "gforce assign high and low took %.3f\n"
-msgstr "gforce ���� high �� low ���� %.3f\n"
+msgstr "gforce 分配 high 和 low 用了 %.3f\n"
```
Glossary 文件里“assign”是翻译成“赋值”了

biobai commented on po/zh_CN.po:
```diff
@@ -3092,48 +3092,48 @@ msgstr ""
 #, c-format
 msgid "gather implemented for INTSXP, REALSXP, and CPLXSXP but not '%s'"
-msgstr "gather ��֧�� INTSXP��REALSXP �� CPLXSXP������֧�� '%s'"
+msgstr "gather 已支持 INTSXP，REALSXP 和 CPLXSXP，但不支持 '%s'"
```
“implemented”此处是不是翻译“执行”？

biobai commented on po/zh_CN.po:
```diff
@@ -3142,133 +3142,133 @@ msgstr ""
-msgstr "GForce ��ͣ�gsum����֧������ '%s'����ʹ�� base::sum(.) ����"
-"���� options(datatable.optimize=1) �ر� GForce �Ż�"
+msgstr "GForce 求和（gsum）不支持类型 '%s'。请使用 base::sum(.) 或者"
+"设置 options(datatable.optimize=1) 关闭 GForce 优化"
```
建议“请使用前缀 base::sum(.) 或者设置 options(datatable.optimize=1) 来关闭 GForce 优化”

biobai commented on po/R-zh_CN.po:
```diff
@@ -1262,43 +1262,51 @@ msgid "length(na.last) > 1, only the first element will be used"
+"input= 必须是以下其中一种字符串: "
+"一个文件名, 一个含有不少于一个空格的系统命令, "
+"以'http[s]://','ftp[s]://' 或 'file://' 开头的URL, "
+"或是包含至少一个\\n 或 \\r的输入数据本身"
```
“或是包含至少一个\\n 或 \\r的输入数据本身” 建议为“或是本身就包含至少一个\\n 或 \\r的输入数据”

biobai commented on po/R-zh_CN.po:
```diff
@@ -1308,18 +1316,22 @@ msgid ""
+"') 同时一个变量也被传输到 `input=`.请使用fread(cmd=...)."
+"这是出于安全考虑. 试想你在开发一个app, 这个app可能有恶意用户, "
+"同时这个app的运行环境不安全 (比如在root环境下运行)."
+"请阅读v1.11.6版本的NEWS file里面的item 5获取更多资讯, 或了解如何减少这个提示."
```
“或了解如何减少这个提示”建议为“或了解如何取消这个提示”

biobai commented on po/R-zh_CN.po:
```diff
@@ -1332,88 +1344,110 @@ msgid ""
 
 msgid "Encountered <"
-msgstr ""
+msgstr "遇到"
```
这里漏掉了“<”

biobai commented:
Hi @Koohoko , I finished the review process. Hope it helps.

Koohoko commented:
> Hi @Koohoko , I finished the review process. Hope it helps.

Thanks so much for your kind help!

Koohoko commented:
@MichaelChirico I agree with most of the comments, but what am I suppose to do to proceed? should I renew the code on my local machine and pull request again? Sorry I dont have experience like this before..

MichaelChirico commented:
Yes, on your local copy, make the edits in the files. Then

```
# check the status of your files locally.
#   only po/R-zh_CN.po and po/zh_CN.po should appear
git status
# if that's working, commit these changes
git commit -am 'updated with feedback'
# push to GitHub; 'origin' is the name where your fork is.
#   You can confirm this by looking at git remote -v
#   origin should be in the line with Koohoko/data.table
git push origin translation-haogao-gu
```

Koohoko commented:
@MichaelChirico My changes are in the "48bdf46" commit. However I seemed to messed up the changes when merging it with the previous edits (which lead to c8fda17).

Basically the "48bdf46" commit contains what I want, and I wish to abandon/revert the c8fda17 commit...

MichaelChirico commented:
I _think_ what happened is you did a merge commit against the current `translation` branch and I _think_ I fixed it by:

 - reset your branch locally to the target commit: `git reset --hard 48bdf46639c204d73d9bb7896c017a7c546f6590`
 - force-push this back to remote: `git push haogao translation-haogao-gu --force`*
 - locally merge the `translation` branch into this branch (`translation-haogao-gu`): `git merge translation`
 - Resolve conflicts -- `po/zh_CN.po` and `po/R-zh_CN.po` were typical, but also two binary .mo files in `inst/po/zh_CN/LC_MESSAGES` were conflicted. I used `git checkout --ours -- inst/po/zh_CN/LC_MESSAGES/*` to "resolve" this conflict, but after finishing the merge, I re-ran `update_pkg_po('.')` and those `.mo` files changed again, so I suspect I should have done `--theirs` instead?

I'll merge now, but please have another glance and check everything looks OK.

* I think this removed some intermediate commits I had added to Haogao's _remote_ branch that hadn't been pulled locally? Or something? I was seeing a message in `git status` about being _5_ commits behind remote after doing `git reset --hard`, when I really should only have been _1_ commit behind. Not sure what those other commits were.

Koohoko commented:
> I _think_ what happened is you did a merge commit against the current `translation` branch and I _think_ I fixed it by:
> 
> * reset your branch locally to the target commit: `git reset --hard 48bdf46639c204d73d9bb7896c017a7c546f6590`
> * force-push this back to remote: `git push haogao translation-haogao-gu --force`*
> * locally merge the `translation` branch into this branch (`translation-haogao-gu`): `git merge translation`
> * Resolve conflicts -- `po/zh_CN.po` and `po/R-zh_CN.po` were typical, but also two binary .mo files in `inst/po/zh_CN/LC_MESSAGES` were conflicted. I used `git checkout --ours -- inst/po/zh_CN/LC_MESSAGES/*` to "resolve" this conflict, but after finishing the merge, I re-ran `update_pkg_po('.')` and those `.mo` files changed again, so I suspect I should have done `--theirs` instead?
> 
> I'll merge now, but please have another glance and check everything looks OK.
> 
> * I think this removed some intermediate commits I had added to Haogao's _remote_ branch that hadn't been pulled locally? Or something? I was seeing a message in `git status` about being _5_ commits behind remote after doing `git reset --hard`, when I really should only have been _1_ commit behind. Not sure what those other commits were.

Thanks @MichaelChirico , I checked the translation in the most recent commit, they appear like what I want. 
I think I have to spend some time learning git, sometimes I am not clear what is happening...
